test/nodes/test_context_optimizer.py

```
@@ -0,0 +1,968 @@
+_install_stubs()
+
+# ---------------------------------------------------------------------------
+# Now import the optimizer -- tiktoken must be available (install or mock).
+# We try a real import first; if unavailable we create a mock.
+# ---------------------------------------------------------------------------
+
+try:
+    import importlib.util
+
+    _TIKTOKEN_AVAILABLE = importlib.util.find_spec('tiktoken') is not None
+except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)